### PR TITLE
PP-5324 Application production logging

### PR DESCRIPTION
**Pull request by sfount**
Default all logs recorded by Toolbox service to include the authenticated user ID - anything HTTP request or internal service API that is accessed with include the session user (ref: security ticket).

* configure JSON production logging
* supplement production logs with request meta data: 
  * current authenticated GitHub username (user id) 
  * correlation ID to track requests through all micro services 
  * toolbox ID to track requests through Toolbox service 
* format development logs to include meta details
* simplify logging in `pay-request` to leverage meta details in favour of custom strings

Micro service tracing: 
* propagate correlation ID through to other microservices, populate the `x-request-id` header if received

### CODE REVIEW

File: src/lib/pay-request/client.js [hunk 1]
```diff
@@ -19,27 +21,48 @@ const serviceApiMethodUtils = require('./api_utils')
...
+    service: this.metadata.serviceKey,
+    method: response.request.method,
+    url: response.config.url,
+    duration: response.config.metadata.duration
+  }
+
   response.config.metadata.end = new Date()
   response.config.metadata.duration = response.config.metadata.end - response.config.metadata.start
-  logger.debug(`[${this.metadata.serviceKey}] "${response.request.method}" success from ${response.config.url} (${response.config.metadata.duration}ms)`)
+  logger.info('Pay request: internal service success fulfilled', logContext)
```
Comment (jankowiakmaria) [inline src/lib/pay-request/client.js]: [nothing major] `success fulfilled` sounds a bit odd to me.

File: src/web/modules/layout/layout.njk [hunk 2]
```diff
@@ -100,6 +100,7 @@
           <div class="govuk-footer__meta-custom">
             <!-- span>pay-toolbox-revised(1.0.4)</span -->
             <!-- span>pay-toolbox-revised(1.0.4) served <code>er32523</code></span -->
+            <span>π ({{ toolboxId }})</span>
```
Comment (jankowiakmaria) [inline src/web/modules/layout/layout.njk]: Why `π `?

Comment (sfount) [reply src/web/modules/layout/layout.njk]: https://en.wikipedia.org/wiki/The_Net_(1995_film)#Plot

File: src/lib/pay-request/client.js (hunk 1 above)
Comment (sfount) [reply src/lib/pay-request/client.js]: You're right, had hoped to rely on context object for this log line but will update the message to include the service and the code for looking from a distance

**Pull request merged**